adds ability to export non-EPSG CRSs URIs

AndrewAnnex opened the pull request:
grabs the authority, version and code from the CRS directly to encode the URIs. Related to #92 but can be merged separately. I'm not aware of any issues this would cause but I am open to hearing about them

AndrewAnnex commented:
okay so I know that the None object is most likely due to the too-high confidence level default to the to_authority call, although I can't seem to find much information on why proj doesn't always have high confidence, is there information missing from the CRS of that polar projection?

AndrewAnnex commented:
okay so although tests pass, I am going to add back a check to ensure we don't try to unpack a None to allow for cases when the right answer really is EPSG/0/None

AndrewAnnex commented on tests/test_models.py:
```diff
@@ -292,8 +292,7 @@ def test_schema():
     json_doc = json.loads(tms.json(exclude_none=True))
-    # We cannot translate PROJ4 to epsg so it's set to None
-    assert json_doc["supportedCRS"] == "http://www.opengis.net/def/crs/EPSG/0/None"
+    assert json_doc["supportedCRS"] == "http://www.opengis.net/def/crs/IAU/2015/49930"
```
okay I can look into that, is there a way to list out all CRSs proj has/all epsg codes?, need to set up a local dev environment to,

AndrewAnnex replied:
ahh found https://pyproj4.github.io/pyproj/stable/api/database.html#pyproj.database.get_codes

however I seem to keep running into issues with setting up pre-commit on macos (typed-ast compilation errors) so this may take a few more commits than is ideal

vincentsarago replied:
I said `all` but it's ok to test only few :D

vincentsarago replied:
do not worry about pre-commit I'll fix formatting later

AndrewAnnex replied:
okay minimal test added to the end of test_models.py

AndrewAnnex commented:
@vincentsarago I've added a test (one parameterized test per authority, not checking every code) and fixed remaining black formatting issues. I could see adding a test to try to ensure that EPSG/0/None is at least returned sometimes, your call.

geospatial-jeff reviewed (approved):
LGTM!
